rust/ql/lib/codeql/rust/dataflow/internal/SsaImpl.qll

```diff
@@ -426,3 +459,44 @@ class PhiReadNode extends DefinitionExt, Impl::PhiReadNode {
 
   override Location getLocation() { result = Impl::PhiReadNode.super.getLocation() }
 }
+
+private module DataFlowIntegrationInput implements Impl::DataFlowIntegrationInputSig {
+  class Expr extends CfgNodes::ExprCfgNode {
+    predicate hasCfgNode(SsaInput::BasicBlock bb, int i) { this = bb.getNode(i) }
+  }
+
+  Expr getARead(Definition def) { result = Cached::getARead(def) }
+
+  /** Holds if SSA definition `def` assigns `value` to the underlying variable. */
+  predicate ssaDefAssigns(WriteDefinition def, Expr value) { none() }
```

Comment:
I think that for Rust we will want `value` to always be the `IdentPat` or `VariableAccess` that underlies `def`. We will then eventually add flow steps that moves values from RHS to LHS in assignments, and read-steps that deconstructs values for patterns.

Reply:
I've made an attempt at implementing `ssaDefAssigns` now. I'm not 100% sure if it's right though.